lesson18/script.js

```diff
@@ -0,0 +1,186 @@
+    const loadingImage = createAttributedElements({tag:"img",attrObj:{src:"img/loading-circle.gif",alt:"ローディング"}});
+    
+    slider.appendChild(loadingPlace).appendChild(loadingImage);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (URL) => {
+    const response = await fetch(URL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage('現在、サーバーの通信が壊れています');
+    }
+    const json = await response.json();
+    const data = json.data;
+    return data;
+}
+
+const fetchSliderData = (ms) => new Promise(resolve => setTimeout(() => resolve(fetchData("https://myjson.dit.upm.es/api/bins/2bzx")),ms));
+
+const init = async () => {
+    showLoading();
+    try{
+        const sliderData = await fetchSliderData(3000);
+        if (!sliderData) {
+            return;
+        }
+        renderSliderElements(sliderData);
+        autoSwitchSlider(sliderData);
+    }catch(error){
+        console.error(error);
+        createErrorMessage(error);
+    }finally{
+        removeLoading();
+    }
+}
+
+const renderSliderElements = (sliderData) => {
+    renderSliderItems(sliderData);
+    getPageNumber(sliderData);
+    renderButtons(sliderData);
+    renderDotPagination(sliderData);
+    isActiveFirstItem();
+}
+
+let autoPlay;
+const autoSwitchSlider = (sliderData) =>{
+    autoPlay = setInterval(() =>{
+        currentIndex ++;
+        if(currentIndex >= sliderData.length){
+            currentIndex = 0;
+        }
+        switchSlider(sliderData,currentIndex);
+    }, 3000);
+}
+
+const resetAutoPlay = (sliderData) => {
+    clearInterval(autoPlay);
+    autoSwitchSlider(sliderData);
+}
+
+const renderSliderItems = (sliderData) => {
+    const fragment = document.createDocumentFragment();
+    for(const image of sliderData){
+        const sliderItem = createAttributedElements({tag:"div",attrObj:{class:"slider__item"}});
+        const slideImage = createAttributedElements({tag:"img",attrObj:{class:"slide-image",src:image.img,alt:image.alt}});
+        
+        fragment.appendChild(sliderItem).appendChild(slideImage);
+    }
+
+    slider.appendChild(fragment);
+}
+
+const getPageNumber = (sliderData) => {
+    const pageNumber    = createAttributedElements({tag:"p",attrObj:{class:"page-number"},str:`/${sliderData.length}`});
+    const currentNumber = createAttributedElements({tag:"span",attrObj:{id:"js-currentNumber",class:"current-number"},str:"1"});
+
+    slider.appendChild(pageNumber).insertAdjacentElement("afterbegin",currentNumber);
+}
+
+const renderButtons = (sliderData) => {
+    const prevBtn = createAttributedElements({tag:"button",attrObj:{id:"js-prevBtn",class:"js-btn prev-btn slider-btn",disabled:true}});
+    const nextBtn = createAttributedElements({tag:"button",attrObj:{id:"js-nextBtn",class:"js-btn next-btn slider-btn"}});
+
+    slider.appendChild(prevBtn).insertAdjacentElement("afterend",nextBtn);
```

Comment:
nits
If you don't need to consider IE, this may be simpler.
```suggestion
    slider.appendChild(prevBtn).after(nextBtn);
```
[ref](https://developer.mozilla.org/ja/docs/Web/API/Element/after)

Reply:
Very simple and nice!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/20/commits/9b5e4afbb08409ce470a1b074958851de968c91f